webui/react/src/hooks/usePermissions.ts

```diff
@@ -0,0 +1,59 @@
+import { useStore } from 'contexts/Store';
+import { Project } from 'types';
+import {
+  canDeleteWorkspaceProjects,
+  canModifyWorkspace,
+  canModifyWorkspaceProjects,
+  canMoveWorkspaceProjects,
+  PermissionWorkspace,
+} from 'utils/role';
+
+interface PermissionsConfig {
+  project?: Project;
+  workspace?: PermissionWorkspace;
+}
+
+type PermissionFn = (arg0: PermissionsConfig) => boolean;
+
+interface PermissionsHook {
```

Comment:
If you wanted to avoid the duplication and don't mind the class baggage you could use a class here.

Reply:
I ended up making a different PermissionsArgs type for the experiments, models, model-versions, projects, workspaces etc.

Reply:
I just meant having to type all the function names twice.